sunpy/map/mapbase.py

```diff
@@ -1542,20 +1542,14 @@ def resample(self, dimensions: u.pixel, method='linear'):
         new_meta = self.meta.copy()
 
         # Update metadata
+        for key in {'cdelt1', 'cd1_1', 'cd2_1'} & self.meta.keys():
+            new_meta[key] *= scale_factor_x
+        for key in {'cdelt2', 'cd1_2', 'cd2_2'} & self.meta.keys():
```

Comment:
Since you acknowledge that it is a "very personal" preference, I'll softly push back.  I feel that the set operators `&` (instead of `.intersection()`), '|' (instead of `.union()`), and `-` (instead of `.difference()`) are sufficiently intuitive to be used without reservation, and certainly much preferred for more complex set expressions (e.g., `(a | b) - (c & d)`).  In contrast, I'm not going to go to bat for `^`, `<`, etc.

Reply:
Certainly for complex operations the brevity wins out. I just think for here if someone who isn't as experience with sets reads this if we use the method then the code "reads" with minimal extra understanding needed. "for key in the intersection of this set with meta keys" where as you can only read it like that with the operator if you know what the operator does.

Reply:
(I am not fussed if we change this or not, I just thought it was a mild readability improvement)